fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

```diff
@@ -0,0 +1,133 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.analyzer;
+
+import com.starrocks.analysis.StatementBase;
+import com.starrocks.analysis.TableName;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.privilege.PrivilegeManager;
+import com.starrocks.privilege.PrivilegeTypes;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.CatalogMgr;
+import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.sql.ast.CTERelation;
+import com.starrocks.sql.ast.JoinRelation;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.ast.SelectRelation;
+import com.starrocks.sql.ast.SetOperationRelation;
+import com.starrocks.sql.ast.SubqueryRelation;
+import com.starrocks.sql.ast.TableRelation;
+import com.starrocks.sql.ast.ViewRelation;
+
+import java.util.Arrays;
+
+public class PrivilegeCheckerV2 {
+
+    private PrivilegeCheckerV2() {}
+
+    public static void check(StatementBase statement, ConnectContext session) {
+        new PrivilegeCheckerVisitor().check(statement, session);
+    }
+
+    public static boolean checkTblPriv(ConnectContext context,
+                                       TableName tableName,
+                                       PrivilegeTypes.TableActions action) {
+        return checkTblPriv(context, tableName.getCatalog(),
+                tableName.getDb(), tableName.getTbl(), action);
+    }
+
+    public static boolean checkTblPriv(ConnectContext context,
+                                       String catalogName,
+                                       String dbName,
+                                       String tableName,
+                                       PrivilegeTypes.TableActions action) {
+        if (!CatalogMgr.isInternalCatalog(catalogName)) {
+            throw new SemanticException("external catalog is not supported for now!");
+        }
+        PrivilegeManager manager = context.getGlobalStateMgr().getPrivilegeManager();
+        // TODO: check catalog & db
```

Comment:
I think adding todo with github id is a better practice in the later codes.

Reply:
Personally, I don't like to add my name in TODO or anywhere else in the code. I prefer to contribute my code silently.
To clarify the author of this TODO, I can always use `git blame`.
To remember what's left undone, I think jira/document is a better choice. Actually I already had a docutment for that.

Reply:
ok, cool~